app/services/medium_providers/provider.rb

```diff
@@ -42,8 +42,8 @@ def additional_media
     end
 
     # Returns unique string representing current installation medium.
-    def unique_id
-      @unique_id ||= Base64.urlsafe_encode64(Digest::SHA1.digest(medium_uri.to_s), padding: false)
+    def unique_id(path = '')
+      @unique_id ||= Base64.urlsafe_encode64(Digest::SHA1.digest(medium_uri(path).to_s), padding: false)
```

Comment:
It's dirty, but the media provider already shares some responsibility with OS. This is a bit fuzzy place to be in, some day we will rearchitect this from scratch? Or not? ;-)

Reply:
One problem is that this breaks the plugin API, apparently Katello needs to add the option too even tho if its unused.